tests/_test_msui/test_mscolab.py

```diff
@@ -344,6 +344,36 @@ def test_view_open(self):
         assert tableview.btAddWayPointToFlightTrack.isEnabled()
         assert any(action.text() == "Ins WP" and action.isEnabled() for action in topview.mpl.navbar.actions())
 
+    def test_multiple_views_and_multiple_flightpath(self):
+        """
+        checks that we can have multiple topviews with the multiple flightpath dockingwidget
+        and we are able to cycle a login/logout
+        """
+        self._connect_to_mscolab()
+        modify_config_file({"MSS_auth": {self.url: self.userdata[0]}})
+        self._login(emailid=self.userdata[0], password=self.userdata[2])
+        # test after activating operation
+        self._activate_operation_at_index(0)
+        self.window.actionTopView.trigger()
+        QtWidgets.QApplication.processEvents()
+        assert len(self.window.get_active_views()) == 1
```

Comment:
Qt is an inherently asynchronous system (at least once signals and slots come into play). This means that there is no guarantee (even with the processEvents call) that all events pertaining to creating the view are already handled once this assert is reached. I.e. there is a potential race condition between this assert and creating the view, which can fail the test.

A better approach is to use pytest-qt's `wait*` methods, specifically `wait_until`: https://pytest-qt.readthedocs.io/en/latest/wait_until.html

I've used it in other locations already to fix potential races that made the test suite unreliable, you can take a look there as well.

This applies to the other asserts as well.